sections/section_util.go

```diff
@@ -0,0 +1,382 @@
+	MspaCoveredTransaction              byte
+	MspaOptOutOptionMode                byte
+	MspaServiceProviderMode             byte
+}
+
+func errorHelper(name string, err error) error {
+	return fmt.Errorf("unable to set field %s due to parse error: %s", name, err.Error())
+}
+
+func NewCommonUSCoreSegment(sensitiveDataFields int, knownChildDataFields int, bs *util.BitStream) (CommonUSCoreSegment, error) {
+	var commonUSCore CommonUSCoreSegment
+	var err error
+
+	commonUSCore.Version, err = bs.ReadByte6()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.Version", err)
+	}
+
+	commonUSCore.SharingNotice, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.SharingNotice", err)
+	}
+
+	commonUSCore.SaleOptOutNotice, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.SaleOptOutNotice", err)
+	}
+
+	commonUSCore.TargetedAdvertisingOptOutNotice, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.TargetedAdvertisingOptOutNotice", err)
+	}
+
+	commonUSCore.SaleOptOut, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.SaleOptOut", err)
+	}
+
+	commonUSCore.TargetedAdvertisingOptOut, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.TargetedAdvertisingOptOut", err)
+	}
+
+	commonUSCore.SensitiveDataProcessing, err = bs.ReadTwoBitField(sensitiveDataFields)
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.SensitiveDataProcessing", err)
+	}
+
+	if knownChildDataFields > 0 {
+		commonUSCore.KnownChildSensitiveDataConsentsArr, err = bs.ReadTwoBitField(knownChildDataFields)
+		if err != nil {
+			return commonUSCore, errorHelper("CoreSegment.KnownChildSensitiveDataConsentsArr", err)
+		}
+	} else {
+		commonUSCore.KnownChildSensitiveDataConsentsInt, err = bs.ReadByte2()
+		commonUSCore.KnownChildSensitiveDataConsentsArr = []byte{}
+		if err != nil {
+			return commonUSCore, errorHelper("CoreSegment.KnownChildSensitiveDataConsentsInt", err)
+		}
+	}
+
+	commonUSCore.MspaCoveredTransaction, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.MspaCoveredTransaction", err)
+	}
+
+	commonUSCore.MspaOptOutOptionMode, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.MspaOptOutOptionMode", err)
+	}
+
+	commonUSCore.MspaServiceProviderMode, err = bs.ReadByte2()
+	if err != nil {
+		return commonUSCore, errorHelper("CoreSegment.MspaServiceProviderMode", err)
+	}
+
+	return commonUSCore, nil
+}
+
+func NewCommonUSGPCSegment(bs *util.BitStream) (CommonUSGPCSegment, error) {
+	var commonUSGPC CommonUSGPCSegment
+	var err error
+
+	commonUSGPC.Gpc, err = bs.ReadByte1()
+	if err != nil {
+		return commonUSGPC, errorHelper("GPCSegment.Gpc", err)
+	}
+
+	return commonUSGPC, nil
+}
+
+func NewUSPCACoreSegment(bs *util.BitStream) (USPCACoreSegment, error) {
```

Comment:
I think we may want to move the non-common initializers into their corresponding packages.

Reply:
Moved these to their respective sections.